### PR TITLE
Duplicate function - divide & conquer

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
The `formatProfileContact` function is a LOT - it is called from the import & causes some notices   - however, as a shared function called by lots of legacy code it isn't something we want to mess with.

We have used the divide and conquer function pretty successfully elsewhere - copy the function back to the class & then with it only being called from one place strip it back to the bit that ACTUALLY DOES something

image

Before
----------------------------------------
Abject fear

After
----------------------------------------
More duplicate code :-( but a path forwards to add import testing & fix on phone imports

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contact/Import/Parser/Contact.php [hunk 1]
```diff
@@ -1555,6 +1555,401 @@ public function createContact(&$formatted, &$contactFields, $onDuplicate, $conta
     return $newContact;
   }
 
+  /**
+   * Legacy format profile contact parameters.
+   *
+   * This is a formerly shared function - most of the stuff in it probably does
+   * nothing but copied here to star unravelling that...
```
Comment (stesi561) [inline CRM/Contact/Import/Parser/Contact.php:1562]: star -> start
:)

**Comment by stesi561** [comment]
Have diffed against the original code - this looks fine. There is a change from using self-> to this-> however the calling is now being done not in a static context so this makes sense. Additionally as the original context was static all the changes to calling methods other than this in the original Class are fine. 

Haven't had a chance to r-run.

**Comment by eileenmcnaughton** [comment]
thanks @stesi561  - there is good test cover here -

File: CRM/Contact/Import/Parser/Contact.php (hunk 1 above)
Comment (eileenmcnaughton) [reply CRM/Contact/Import/Parser/Contact.php:1562]: Hmm - I might take that on board for a follow up as I have a few patches locally that build on this

**Pull request merged**